test/token/ERC721/extensions/ERC721Royalty.test.js

```diff
@@ -0,0 +1,114 @@
+const { BN, constants, expectRevert } = require('@openzeppelin/test-helpers');
+const { expect } = require('chai');
+const { ZERO_ADDRESS } = constants;
+
+const { shouldSupportInterfaces } = require('../../../utils/introspection/SupportsInterface.behavior');
+
+const ERC721RoyaltyMock = artifacts.require('ERC721RoyaltyMock');
+
+contract('ERC721Royalty', function (accounts) {
+  const [ account1 ] = accounts;
+  const tokenId1 = new BN('1');
+  const tokenId2 = new BN('2');
+  const salePrice = new BN('1000');
+  const royaltyAmount = new BN('10');
+
+  beforeEach(async function () {
+    this.token = await ERC721RoyaltyMock.new();
+  });
+
+  shouldSupportInterfaces(['ERC2981']);
+
+  describe('global royalty', function () {
+    beforeEach(async function () {
+      await this.token.setRoyalty(account1, royaltyAmount);
+    });
+
+    it('updates royalty amount', async function () {
+      const newAmount = new BN('25');
+      let royalty = new BN((salePrice * royaltyAmount) / 100);
+      // Initial royalty check
+      const initInfo = await this.token.royaltyInfo(tokenId1, salePrice);
+
+      expect(initInfo.receiver).to.be.equal(account1);
+      expect(initInfo.royaltyAmount).to.be.bignumber.equal(royalty);
+
+      // Updated royalty check
+      await this.token.setRoyalty(account1, newAmount);
+      royalty = new BN((salePrice * newAmount) / 100);
+      const newInfo = await this.token.royaltyInfo(tokenId1, salePrice);
+
+      expect(newInfo.receiver).to.be.equal(account1);
+      expect(newInfo.royaltyAmount).to.be.bignumber.equal(royalty);
+    });
+
+    it('holds same royalty value for different tokens', async function () {
+      const newAmount = new BN('20');
+      await this.token.setRoyalty(account1, newAmount);
+
+      const token1Info = await this.token.royaltyInfo(tokenId1, salePrice);
+      const token2Info = await this.token.royaltyInfo(tokenId2, salePrice);
+
+      expect(token1Info.royaltyAmount).to.be.bignumber.equal(token2Info.royaltyAmount);
+    });
+
+    it('reverts if invalid parameters', async function () {
+      await expectRevert(
+        this.token.setRoyalty(ZERO_ADDRESS, royaltyAmount),
+        'ERC2981: Invalid recipient',
+      );
+
+      await expectRevert(
+        this.token.setRoyalty(account1, new BN('100')),
```

Comment:
Does the EIP disallow a 100% royalty amount? It's a bit ridiculous maybe but I think it should be allowed. Unless the EIP says otherwise, I personally think the royalty percentage could be set to >100%... HOWEVER, as a sanity check and general protection I think our implementation should not allow >100%.